xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-wikimacro/xwiki-platform-rendering-wikimacro-api/src/main/java/org/xwiki/rendering/macro/wikibridge/binding/WikiMacroBinding.java

```diff
@@ -90,7 +90,12 @@ public String getContent()
         return (String) this.get(CONTENT);
     }
 
-    protected void setContent(String content)
+    /**
+     * Set the content of the macro.
+     *
+     * @param content the new macro content
+     */
+    public void setContent(String content)
```

Comment:
The use of setContent should be limited to wiki macro authors with programming right as otherwise a wiki macro could use the right of whoever uses it (since the content is executed with the right of its author).